core/internal/pathtree/pathtree.go

```diff
@@ -53,79 +49,87 @@ func (pt *PathTree) CloneTree() (TreeData, error) {
 	return clone, nil
 }
 
-// Updates and/or removes values from the tree.
+// Set changes the value of the leaf node at the given path.
 //
-// Does a best-effort job to apply all changes. Errors are passed to `onError`
-// and skipped.
-func (pt *PathTree) ApplyUpdate(
-	items []*PathItem,
-	onError func(error),
-) {
-	for _, item := range items {
-		if err := updateAtPath(pt.tree, item.Path, item.Value); err != nil {
-			onError(err)
-			continue
-		}
-	}
-}
+// If the path doesn't refer to a node in the tree, nodes are inserted
+// and a new leaf is created.
+//
+// If path refers to a non-leaf node, that node is replaced by a leaf
+// and the subtree is discarded.
+func (pt *PathTree) Set(path TreePath, value any) {
```

Comment:
nit: how about Insert? or would be wrong in case of replacement?

Reply:
Insert is commonly used in list data structures to refer to a nondestructive operation. Since this can change existing data, Set is more appropriate.